rstudio/rstudio
# add Dockerfile editor mode

### Intent

Bring in basic editor support for Dockerfiles.

### Approach

Ace already provides an editor mode + highlight rules for Dockerfiles, so just define an editor mode and use that.

### QA Notes

Open a sample Dockerfile, and confirm that basic syntax highlighting is present.

<img width="398" alt="Screen Shot 2020-10-09 at 2 35 22 PM" src="https://user-images.githubusercontent.com/1976582/95633528-b2434280-0a3c-11eb-9c44-c42e59be4b60.png">

Note that we definitely don't intend to provide any higher-level features at this point; this is purely to allow users to quickly edit Dockerfiles as necessary from RStudio.

---

Closes https://github.com/rstudio/rstudio/issues/5141.
